server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -345,9 +347,33 @@ private <C extends Collector, T> T search(Weight weight, CollectorManager<C, T>
 
     @Override
     public void search(List<LeafReaderContext> leaves, Weight weight, Collector collector) throws IOException {
-        collector.setWeight(weight);
-        for (LeafReaderContext ctx : leaves) { // search each subreader
-            searchLeaf(ctx, weight, collector);
+        try {
+            collector.setWeight(weight);
+            for (LeafReaderContext ctx : leaves) { // search each subreader
+                searchLeaf(ctx, weight, collector);
+            }
+        } catch (@SuppressWarnings("unused") TimeExceededException e) {
```

Comment:
does this cover all the cases where exitable directory reader may also throw Time exceeded exception, or only for the cases where cancellable bulk scorer throws it? This is somehow different from current lucene timeout support.

Reply:
No, as I said in https://github.com/elastic/elasticsearch/pull/97771#issuecomment-1641422849, we will need to wrap the creation of the weight as it can throw a `TimeExceededException`.

Reply:
I think I cover it in the last commit